catalog/confluence/catalog-confluence-source/src/main/java/org/codice/ddf/confluence/source/ConfluenceFilterDelegate.java

```diff
@@ -0,0 +1,243 @@
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.codice.ddf.confluence.common.Confluence;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.types.Contact;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.data.types.Topic;
+import ddf.catalog.filter.impl.SimpleFilterDelegate;
+
+public class ConfluenceFilterDelegate extends SimpleFilterDelegate<String> {
+
+    public static final Map<String, ConfluenceQueryParameter> QUERY_PARAMETERS;
+
+    private static final String PROPERTY_FORMAT = "%s %s \"%s\"";
+
+    private static final String DATE_FORMAT = "yyyy-MM-dd HH:mm";
+
+    private static final String WILD_CARD = "*";
+
+    private boolean wildcardQuery = false;
+
+    private int parameterCount = 0;
+
+    private Set<String> queryTags = new HashSet<>();
+
+    static {
+        Map<String, ConfluenceQueryParameter> params = new HashMap<>();
+        params.put(Metacard.ID,
+                new ConfluenceQueryParameter("id", false, true, false, false, true));
+        params.put(Metacard.CONTENT_TYPE,
+                new ConfluenceQueryParameter("type", false, true, false, false, true));
+        params.put(Metacard.TITLE,
+                new ConfluenceQueryParameter("title", true, true, false, true, false));
+        params.put(Metacard.ANY_TEXT,
+                new ConfluenceQueryParameter("text", true, false, false, true, false));
+        params.put(Metacard.CREATED,
+                new ConfluenceQueryParameter("created", false, true, true, false, false));
+        params.put(Metacard.MODIFIED,
+                new ConfluenceQueryParameter("lastmodified", false, true, true, false, false));
+        params.put(Topic.CATEGORY,
+                new ConfluenceQueryParameter("type", false, true, false, false, true));
+        params.put(Contact.CREATOR_NAME,
+                new ConfluenceQueryParameter("creator", false, true, false, true, true));
+        params.put(Contact.CONTRIBUTOR_NAME,
+                new ConfluenceQueryParameter("contributor", false, true, false, true, true));
+        params.put(Topic.KEYWORD,
+                new ConfluenceQueryParameter("label", false, true, false, false, true));
+        params.put(Core.METACARD_CREATED,
+                new ConfluenceQueryParameter("created", false, true, true, false, false));
+        params.put(Core.METACARD_MODIFIED,
+                new ConfluenceQueryParameter("lastmodified", false, true, true, false, false));
+        QUERY_PARAMETERS = Collections.unmodifiableMap(params);
+
+    }
+
+    @Override
+    public String and(List<String> operands) {
+        return operands.stream()
+                .filter(Objects::nonNull)
+                .collect(Collectors.joining(" AND ", "( ", " )"))
+                .replaceAll("\\(\\s*\\)", "");
```

Comment:
You might want to extract this common pattern as a constant.  Then you could name it EMPTY_GROUP_PATTERN or something like that so it is more obvious what the replaceAll is doing everywhere.